### PR TITLE
Disable caching for MergeNativeLibs.

### PR DESCRIPTION
**Pull request by runningcode**
This disables caching for the MergeNativeLibs task because it behaves like a copy task which do not benefit from caching.

This PR also removes the workaround for DexFileDependencies since this task is not cacheable by default.

### CODE REVIEW

File: src/main/groovy/org/gradle/android/workarounds/MergeNativeLibsWorkaround.groovy [hunk 1]
```diff
@@ -2,37 +2,19 @@ package org.gradle.android.workarounds
...
-    String propertyName = "projectNativeLibs"
+import org.gradle.api.Project
 
+@AndroidIssue(introducedIn = "3.5.0", fixedIn = [], link = "https://issuetracker.google.com/issues/153088766")
+class MergeNativeLibsWorkaround implements Workaround {
     @Override
     void apply(WorkaroundContext context) {
-        super.apply(context)
+        context.project.tasks.withType(MergeNativeLibsTask).configureEach {
+            outputs.doNotCacheIf("Caching MergeNativeLibs adds overhead to the build and never saves time", {true })
```
Comment (ghale) [inline src/main/groovy/org/gradle/android/workarounds/MergeNativeLibsWorkaround.groovy:12]: This feels a little too strongly worded.  Perhaps "Caching MergeNativeLibs is unlikely to provide positive performance results".

**Comment by etiennestuder** [comment]
I'd like for you and @ghale to decide.

**Review by ghale — CHANGES_REQUESTED** [review]
Can you provide more color as to why `DexFileDependenciesWorkaround` is being removed?  Is this one where we've found that it doesn't provide significant gains?

File: src/main/groovy/org/gradle/android/workarounds/MergeNativeLibsWorkaround.groovy (hunk 1 above)
Comment (runningcode) [reply src/main/groovy/org/gradle/android/workarounds/MergeNativeLibsWorkaround.groovy]: Updated!

**Review by ghale — APPROVED** [review]

**Pull request merged**